docs/nd4j/templates/quickstart.md

```diff
@@ -0,0 +1,381 @@
+
+ [[   12.0000,   13.0000,   14.0000,   15.0000], 
+  [   16.0000,   17.0000,   18.0000,   19.0000], 
+  [   20.0000,   21.0000,   22.0000,   23.0000]]]
+*/
+```
+
+### Basic Operations
+You will have to use INDArray methods to perform operations on your arrays. There are  in-place and copy overloads and scalar and element wise overloaded versions. The in-place operators return a reference to the array so you can conveniently chain operations together.
+
+```java
+//Copy
+arr_new = arr.add(scalar);    // return a new array with scalar added to each element of arr.
+arr_new = arr.add(other_arr); // return a new array with element wise addition of arr and other_arr.
+
+//in place.
+arr_new = arr.addi(scalar); //Heads up: arr_new points to the same array as arr.
+arr_new = arr.addi(other_arr);
+```
+
+addition: arr.add(...), arr.addi(...)
+substraction: arr.sub(...), arr.subi(...)
+multiplication: arr.mul(...), arr.muli(...)
+division: arr.div(...), arr.divi(...)
+
+When you perform the basic operations you must make sure the underlying data types are the same.
+```java
+int [] shape = {5};
+INDArray  x = Nd4j.zeros(shape, DataType.DOUBLE);
+INDArray  x2 = Nd4j.zeros(shape, DataType.INT);
+INDArray  x3 = x.add(x2);
+// java.lang.IllegalArgumentException: Op.X and Op.Y must have the same data type, but got INT vs DOUBLE
+```
+<!--- Moving matrix operations after Transforms. In Nd4j the dot product is a transform. -->
+
+The IndArray has methods implementing operations such as `sum`, `min`, `max`.
+```java
+int [] shape = {2,3};
+INDArray  x = Nd4j.rand(shape);
+System.out.println(x);
+
+System.out.println(x.sum());
+System.out.println(x.min());
+System.out.println(x.max());
+/*
+[[    0.8621,    0.9224,    0.8407], 
+ [    0.1504,    0.5489,    0.9584]]
+4.2830
+0.1504
+0.9584
+*/
+```
+
+Provide a dimension argument to apply the operation across the specified dimension:
+
+```java
+int [] shape = {3,4};
+INDArray x = Nd4j.arange(12).reshape(shape);
+System.out.println(x);
+/*
+[[         0,    1.0000,    2.0000,    3.0000], 
+ [    4.0000,    5.0000,    6.0000,    7.0000], 
+ [    8.0000,    9.0000,   10.0000,   11.0000]]
+*/        
+
+System.out.println(x.sum(0)); // Sum of each column.
+//[   12.0000,   15.0000,   18.0000,   21.0000]
+
+System.out.println(x.min(1)); // Min of each row
+//[         0,    4.0000,    8.0000]
+
+System.out.println(x.cumsum(1)); // cumulative sum across each row,
+/*
+[[         0,    1.0000,    3.0000,    6.0000], 
+ [    4.0000,    9.0000,   15.0000,   22.0000], 
+ [    8.0000,   17.0000,   27.0000,   38.0000]]
+*/
+
+```
+
+<!--- The numpy quickstart calls them Universal Functions. -->
+### Transform operation
+Nd4j provides familiar mathematical functions such as sin, cos, and exp. These are called transform operations. The result is returned as an INDArray.
+
+```java
+INDArray x = Nd4j.arange(3);
+System.out.println(x);
+// [         0,    1.0000,    2.0000]
+System.out.println(exp(x));
```

Comment:
I'd rather we don't rely on static imports here. It's nice and short, but user will copy/paste this, miss the static import, and then be confused.
```
import org.nd4j.linalg.ops.transforms.Transforms
System.out.println(Transforms.exp(x));
```

Reply:
I will stick the imports on top.